internal/datastore/datastore.go

```diff
@@ -53,13 +59,32 @@ type Datastore interface {
 	UpsertUserDevice(ctx context.Context, userDevice model.UserDevice) (*model.UserDevice, error)
 	GetViewersByIDs(ctx context.Context, viewerIDs []string) (map[string]*model.Viewer, error)
 	UpsertViewer(ctx context.Context, viewer model.Viewer) (*model.Viewer, error)
+	GetPostByID(ctx context.Context, postID string) (*model.Post, error)
 }
 
 type delphisDB struct {
 	dynamo   dynamodbiface.DynamoDBAPI
 	sql      *gorm.DB
+	pg       *sql2.DB
 	dbConfig config.TablesConfig
 	encoder  *dynamodbattribute.Encoder
+
+	// Prepared Statements
+	// Post
+	putPostStmt                *sql2.Stmt
```

Comment:
Can we put prepared statements in a different struct? Will make dependency injection slightly easier, I think.